src/NumericalAlgorithms/LinearOperators/Linearize.cpp

```diff
@@ -6,39 +6,41 @@
 #include <functional>
 
 #include "DataStructures/DataVector.hpp"
-#include "DataStructures/Index.hpp"  // IWYU pragma: keep
 #include "DataStructures/Matrix.hpp"
+#include "DataStructures/Mesh.hpp"  // IWYU pragma: keep
 #include "NumericalAlgorithms/LinearOperators/ApplyMatrices.hpp"
-#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/MakeArray.hpp"
-#include "Utilities/StdArrayHelpers.hpp"
 
 template <size_t Dim>
-DataVector linearize(const DataVector& u, const Index<Dim>& extents) {
-  const auto linear_filter_matrices =
-      map_array(extents.indices(), [](const size_t extent) noexcept {
-        return std::cref(Basis::lgl::linear_filter_matrix(extent));
-      });
-  return apply_matrices(linear_filter_matrices, u, extents);
+DataVector linearize(const DataVector& u, const Mesh<Dim>& mesh) {
+  const Matrix empty{};
```

Comment:
Why this change?

Reply:
The extents are not sufficient anymore for the call to get the matrix, instead it needs `mesh.slice_through(d)` for `d=0..Dim`. Do you have a suggestion on how to do this in a more concise implementation?